docs/DeveloperGuide.md

```diff
@@ -242,6 +242,40 @@ that NUS course is suitable to be mapped overseas in South East Asia and Oceania
 #### Sequence Diagram:
 ![Filter Courses Sequence Diagram](images/FilterCoursesCommand.png)
 
+### 7. Add Courses Command
+
+#### Overview:
+This command is responsible for adding users' desired course mapping into the `myList.json` file.
+Additionally, each course mapping is checked against the current course mapping found in the
+`database.json` file, ensuring that the course mapping is accurate and is limited to Oceania
+and South-East Asian universities. This command hence helps the users to keep track of their course mapping process.
+
+#### How the feature is implemented:
+* The `AddCoursesCommand` class extends `Command` class where it overrides the `execute` method for
+  custom behaviour.
+* The command first reads a JSON file to obtain the names via `createJsonObject()` method from the
+  superclass.
+* The `trimString` method then removes the `add` command and checks whether the user gave any input after the command.
+  The method would return the user's input without the command.
+* This input is then passed into the `parseAddCommand()` method to obtain the relevant information: NUS course code,
+  name of partner university and partner university course code.
```

Comment:
Will you fix the command summary in another PR btw? or in here together? @hzxnancy